scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5159 @@
+        By definition of the support, the log-PDF evaluates to its minimum value
+        of :math:`-\infty` (i.e. :math:`\log(0)`) outside the support; i.e. for
+        :math:`x < l` or :math:`x > r`. The maximum of the log-PDF may be less
+        than or greater than :math:`\log(1) = 0` because the maximum of the PDF
+        can be any positive real.
+
+        For distributions with infinite support, it is common for
+        `pdf` to return a value of ``0`` when the argument
+        is theoretically within the support; this can occur because the true value
+        of the PDF is too small to be represented by the chosen dtype. The log
+        of the PDF, however, will often be finite (not ``-inf``) over a much larger
+        domain. Consequently, it may be preferred to work with the logarithms of
+        probabilities and probability densities to avoid underflow.
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> import numpy as np
+        >>> from scipy import stats
+        >>> X = stats.Uniform(a=-1.0, b=1.0)
+
+        Evaluate the log-PDF at the desired argument:
+
+        >>> X.logpdf(0.5)
+        -0.6931471805599453
+        >>> np.allclose(X.logpdf(0.5), np.log(X.pdf(0.5)))
+        True
+
+        """
+        return self._logpdf_dispatch(x, method=method, **self._parameters)
+
+    @_dispatch
+    def _logpdf_dispatch(self, x, *, method=None, **kwargs):
+        if self._overrides('_logpdf_formula'):
+            method = self._logpdf_formula
+        elif self.tol is _null:  # ensure that developers override _logpdf
+            method = self._logpdf_logexp
+        return method
+
+    def _logpdf_formula(self, x, **kwargs):
+        raise NotImplementedError(self._not_implemented)
+
+    def _logpdf_logexp(self, x, **kwargs):
+        return np.log(self._pdf_dispatch(x, **kwargs))
+
+    @_set_invalid_nan
+    def pdf(self, x, *, method=None):
+        r"""Probability density function
+
+        The probability density function :math:`f(x)` is the probability *per
+        unit interval* of :math:`x` that the random variable takes on the value
+        :math:`x`. Mathematically, it can be defined as the derivative of the
+        cumulative distribution function `F(x)`:
+
+        .. math::
+
+            f(x) = \frac{d}{dx} F(x)
+
+        `pdf` accepts `x` for :math:`x`.
+
+        Parameters
+        ----------
+        x : Array
+            The value of the random variable for which the PDF is to be
+            evaluated.
+        method : {None, 'formula', 'logexp'}
+            The strategy used to evaluate the PDF. By default (``None``), the
+            infrastructure chooses between the following options, listed in
+            order of precedence.
+
+            - ``'formula'``: use a formula for the PDF itself
+            - ``'logexp'``: evaluate the logarithm of the PDF directly and
+                            exponentiate
```

Comment:
Consider renaming to logpdf to match the method name

Reply:
The idea was to use the same names in all functions. In PDF, `logexp` means exponentiate the log PDF. It's always the same idea - use the log/exponential relationship - so I'd prefer not to have a different value to remember for each function.

*Technically where I wrote function above should be method, as in instance method. Word changed to avoid ambiguity.*